homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,178 @@
+            except FSConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception as exception:  # pylint: disable=broad-except
+                _LOGGER.exception(exception)
+                errors["base"] = "unknown"
+            else:
+                return await self._async_step_device_config_if_needed()
+
+        data_schema = self.add_suggested_values_to_schema(
+            STEP_USER_DATA_SCHEMA, user_input
+        )
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def _async_step_device_config_if_needed(self) -> FlowResult:
+        """Most users will not have changed the default PIN on their radio.
+
+        We try to use this default PIN, and only if this fails ask for it via `async_step_device_config`
+        """
+
+        try:
+            # try to login with default pin
+            afsapi = AFSAPI(self._webfsapi_url, DEFAULT_PIN)
+
+            self._name = await afsapi.get_friendly_name()
+        except InvalidPinException:
+            # Ask for a PIN
+            return await self.async_step_device_config()
+
+        self.context["title_placeholders"] = {"name": self._name}
+
+        self._unique_id = await afsapi.get_radio_id()
+        await self.async_set_unique_id(self._unique_id)
+        self._abort_if_unique_id_configured()
+
+        return await self._create_entry()
+
+    async def async_step_device_config(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle device configuration step.
+
+        We ask for the PIN in this step.
+        """
+        assert self._webfsapi_url is not None
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="device_config", data_schema=STEP_DEVICE_CONFIG_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            afsapi = AFSAPI(self._webfsapi_url, user_input[CONF_PIN])
+
+            self._name = await afsapi.get_friendly_name()
+
+        except FSConnectionError:
+            errors["base"] = "cannot_connect"
+        except InvalidPinException:
+            errors["base"] = "invalid_auth"
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            errors["base"] = "unknown"
+        else:
+            self._unique_id = await afsapi.get_radio_id()
+            await self.async_set_unique_id(self._unique_id)
+            self._abort_if_unique_id_configured()
+            return await self._create_entry(pin=user_input[CONF_PIN])
+
+        data_schema = self.add_suggested_values_to_schema(
+            STEP_DEVICE_CONFIG_DATA_SCHEMA, user_input
+        )
+        return self.async_show_form(
+            step_id="device_config",
+            data_schema=data_schema,
+            errors=errors,
+        )
+
+    async def _create_entry(self, pin: str | None = None) -> FlowResult:
```

Comment:
IMHO this method adds complexity that is not needed.
Instead of a simple `async_create_entry` call, we not have additional variables and calls over the place (that are out of standard).

I suggest to just remove this method fully, as the only thing it does is making a single call.